### ISSUE TITLE
Compatibility check task sometimes failing due to find configuration/Buf module for input

### ISSUE THREAD

**Issue by dkichler-improving** [improving-ottawa/sbt-buf#10]
Still unclear why.  Currently has only occurred in the context of a pipeline build, have not observed this running the task locally.

Stacktrace:
```
[error] java.lang.IllegalStateException: Cannot determine module proto src Buf module to use for input configuration of compatibility check
[error] 	at com.yoppworks.sbt.SbtBufPlugin$.$anonfun$runBufCompatCheck$7(SbtBufPlugin.scala:298)
[error] 	at scala.Option.getOrElse(Option.scala:189)
[error] 	at com.yoppworks.sbt.SbtBufPlugin$.$anonfun$runBufCompatCheck$3(SbtBufPlugin.scala:297)
[error] 	at com.yoppworks.sbt.SbtBufPlugin$.$anonfun$runBufCompatCheck$3$adapted(SbtBufPlugin.scala:279)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:829)
[error] (api / bufCompatCheck) java.lang.IllegalStateException: Cannot determine module proto src Buf module to use for input configuration of compatibility check
```

**Comment by dkichler-improving** [comment]
Looks like a legitimate race condition between `bufCompatCheck` and `generateBufFiles` because there is no explicit dependency in the task graph.

**Comment by dkichler-improving** [comment]
Addressed by #12
